parquet-column/src/main/java/parquet/example/data/simple/convert/SimplePrimitiveConverter.java

```diff
@@ -18,13 +18,18 @@
  */
 package parquet.example.data.simple.convert;
 
+import parquet.column.Dictionary;
+import parquet.example.data.simple.Primitive;
+import parquet.io.ParquetDecodingException;
 import parquet.io.api.Binary;
 import parquet.io.api.PrimitiveConverter;
+import parquet.io.api.RecordConsumer;
 
 class SimplePrimitiveConverter extends PrimitiveConverter {
 
   private final SimpleGroupConverter parent;
   private final int index;
+  private DictionaryPrimitive[] dictionaryPrimitives;
```

Comment:
I think this was Ryan's point. I need to add abstract method to Dictionary, and probably a couple of new dictionaries covering Int96 and FixedLenByteArray